dotnet/src/InternalUtilities/connectors/AI/FunctionCalling/FunctionCallsProcessor.cs

```diff
@@ -265,27 +231,203 @@ public FunctionCallsProcessor(ILogger? logger = null)
         return null;
     }
 
+    /// <summary>
+    /// Processes function calls specifically for Open AI Assistant API.  In this context, the chat-history is not
+    /// present in local memory.
+    /// </summary>
+    /// <param name="chatMessageContent">The chat message content representing AI model response and containing function calls.</param>
+    /// <param name="checkIfFunctionAdvertised">Callback to check if a function was advertised to AI model or not.</param>
+    /// <param name="options">Function choice behavior options.</param>
+    /// <param name="kernel">The <see cref="Kernel"/>.</param>
+    /// <param name="isStreaming">Boolean flag which indicates whether an operation is invoked within streaming or non-streaming mode.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
+    /// <returns>Last chat history message if function invocation filter requested processing termination, otherwise null.</returns>
+    public async IAsyncEnumerable<FunctionResultContent> InvokeFunctionCallsAsync(
+        ChatMessageContent chatMessageContent,
+        Func<FunctionCallContent, bool> checkIfFunctionAdvertised,
+        FunctionChoiceBehaviorOptions options,
+        Kernel kernel,
+        bool isStreaming,
+        [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        FunctionCallContent[] functionCalls = FunctionCallContent.GetFunctionCalls(chatMessageContent).ToArray();
+
+        this._logger.LogFunctionCalls(functionCalls);
+
+        List<Task<FunctionResultContext>> functionTasks = new(functionCalls.Length);
+
+        // We must send back a result for every function call, regardless of whether we successfully executed it or not.
+        // If we successfully execute it, we'll add the result. If we don't, we'll add an error.
+        for (int functionCallIndex = 0; functionCallIndex < functionCalls.Length; functionCallIndex++)
+        {
+            FunctionCallContent functionCall = functionCalls[functionCallIndex];
+            ChatMessageContent functionCallMessage =
+                new()
+                {
+                    Items = [.. functionCalls]
+                };
```

Comment:
Do we necessarily need to create the new message for every function call? Can't we reuse the original message that contains the function calls?

Reply:
Yes, I changed my initial approach and am passing in a message.  Missed this update.